### DISCUSSION TITLE
Further jtag2updi improvement

### DISCUSSION THREAD

**Discussion by mcuee** [avrdudes/avrdude#1678] [Ideas]
There are some suggestions from @askn37 to improve jtag2updi. He has also patched FW for his jtag2updi implementation.

**Comment by mcuee** [comment]
To do: to mention about the improvements which are good to have.

**Comment by mcuee** [comment]
Existing issue #1661 which was fixed by #1662.
* https://github.com/avrdudes/avrdude/issues/1661
* https://github.com/avrdudes/avrdude/issues/1662

**Comment by mcuee** [comment]
@askn37's improved jtag2updi firmware
https://github.com/askn37/jtag2updi/

**Comment by mcuee** [comment]
@MCUdude 
The above `jtag2updi` fork supports to be installed on older AVRs, and it suports new UPDI AVRs, similar to `serialupdi`.

> Equipment that can be installed with firmware:
>
> ATmegaX8 series --- Arduino UNO Rev.3, Arduino Mini, Arduino Micro (ATmega328P)
> megaAVR-0 series --- Arduino UNO WiFi Rev.2 (ATmega4809)
> AVR_DA/DB Series --- Microchip Curiocity
>
> - Compatible with NVMCTRL version 0,2,3,4,5. The correspondence in the original text is up to 0 and 2.. Support tinyAVR-0/1/2, megaAVR-0, AVR_DA/DB/DD/EA/EB.
> - Compatible with AVRDUDE 7.3

I will try the above using Arduno Uno Rev 3 clones.

**Comment by mcuee** [comment]
@askn37 has another repo but that can not be installed on the older AVRs.
https://github.com/askn37/multix-zinnia-updi4avr-firmware-builder/

**Comment by stefanrueger** [comment]
Closing this discussion for the time being. New issues with specific improvements to AVRDUDE's code base are welcome. In particular, @askn37 would be welcome to contribute.
